controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteStreamTask.java

```diff
@@ -81,6 +83,18 @@ public DeleteStreamTask(final StreamMetadataTasks streamMetadataTasks,
                 });
     }
 
+    private CompletableFuture<Void> deleteAssociatedStreams(String scope, String stream, long requestId) {
+        String markStream = NameUtils.getMarkStreamForStream(stream);
+        OperationContext context = streamMetadataStore.createContext(scope, markStream);
+        return streamMetadataStore.checkStreamExists(scope, markStream)
+            .thenCompose(exists -> {
+                if (exists) {
+                    return notifyAndDelete(context, scope, markStream, requestId);
```

Comment:
Can notifyAndDelete be made idempotent so this exists check is not needed?

Reply:
its actually idempotent, except it can throw DataNotFoundException if metadata is not found. 
i am catching that now instead of doing checkExists. 
Also, that uncovered a bug in InMemoryStream where for one case we do not throw DataNotFound which can cause NPE. Fixed that too.